app/jobs/csv_export_job.rb

```diff
@@ -34,28 +34,44 @@ def deploy_csv_export(csv_export)
     filename = csv_export.path_file
     filter = csv_export.filters
 
-    @deploys = Deploy.joins(:stage, :job).where(filter)
+    get_deploys(filter)
     summary = ["-", "Generated At", csv_export.updated_at, "Deploys", @deploys.count.to_s]
     filters_applied = ["-", "Filters", filter.to_json]
 
     CSV.open(filename, 'w+') do |csv|
-      csv << [
-        "Deploy Number", "Project Name", "Deploy Sumary", "Deploy Commit", "Deploy Status", "Deploy Updated",
-        "Deploy Created", "Deployer Name", "Deployer Email", "Buddy Name", "Buddy Email",
-        "Production Flag", "No code deployed"
-      ]
+      csv << Deploy.csv_header
       @deploys.find_each do |deploy|
-        csv << [
-          deploy.id, deploy.project.name, deploy.summary, deploy.commit, deploy.job.status, deploy.updated_at,
-          deploy.start_time, deploy.job.user.name, deploy.job.user.try(:email), deploy.buddy_name, deploy.buddy_email,
-          deploy.stage.production, deploy.stage.no_code_deployed
-        ]
+        csv << deploy.csv_line
       end
       csv << summary
       csv << filters_applied
     end
   end
 
+  def get_deploys(filter)
+    if filter.keys.include?('environments.production')
```

Comment:
so this is the `searching for deleted` test ? ... can we make that more obvious ? :)